### PR TITLE
KOGITO-7232: Push GH action on main to deploy on GH pages

### PR DESCRIPTION
**Pull request by Ginxo**
This Github action is run on every pull request merged into the main branch. This action will upload the build artifacts to gh-pages branch. Github pages will render this content.

### CODE REVIEW

**Comment by ricardozanini** [comment]
@Ginxo mind sharing a description of what this PR does?

**Comment by Ginxo** [comment]
sure @ricardozanini . It was the output from a meeting we had this morning with @akumar074 and @amadhusu 
I guess they can provide more details.

**Comment by akumar074** [comment]
@ricardozanini This came up after we had a discussion regarding publishing our app to GH pages. This PR is WIP, and this helps us to create a pipeline for the latest documentation.

**Comment by ricardozanini** [comment]
Thanks, guys! When opening a PR, please try adding a description to it or a linked JIRA. Sorry to be a PITA, but we can follow the great work you are already doing this way! 👍

**Comment by akumar074** [comment]
Hello @Ginxo , @ricardozanini 
I updated the action to use a prebuilt [action recommended by Antora](https://docs.antora.org/antora/latest/publish-to-github-pages/). This action would overcome a few drawbacks of gh-pages such as Jekyll is run by default on the GH pages which makes it hinders deploying Antora sites. Please take a look at the PR and share your thoughts.
Thanks :)

**Review by amadhusu — APPROVED** [review]
Tested the outcome. Hence Approved !!!!

File: .github/workflows/npm-publish-pages.yml [hunk 1]
```diff
@@ -4,26 +4,34 @@
 name: Node.js Publish webpage
 
 on:
-  release:
-    types: [created]
+  push:
+    branches: [ "main" ]
```
Comment (Ginxo) [inline .github/workflows/npm-publish-pages.yml:8]: I would recommend to use
```
on:
  push:
    branches:
      - main
```

**Review by Ginxo — COMMENTED** [review]
LGTM, just a comment

File: .github/workflows/npm-publish-pages.yml (hunk 1 above)
Comment (akumar074) [reply .github/workflows/npm-publish-pages.yml]: Thanks @Ginxo ! I have updated the file.

**Comment by akumar074** [comment]
@ricardozanini Once we merge this PR, we can update the links to serverless workflow guides in Kogito-website. Right? or should we wait for all the content to be available.

**Review by hmanwani-rh — APPROVED** [review]

**Comment by ricardozanini** [comment]
> @ricardozanini Once we merge this PR, we can update the links to serverless workflow guides in Kogito-website. Right? or should we wait for all the content to be available.

Yes, let's publish already. It's an added motivation for SMEs to keep writing. We already have a few in the queue.

**Review by akumar074 — APPROVED** [review]

**Comment by akumar074** [comment]
Hello guys! Thanks for taking a look. I'm merging this PR and will test the action.

**Comment by akumar074** [comment]
After push, we got an error while running `npm ci` command. It looks like `package-lock.json` is removed from the main which is mandatory. `npm ci` ensures a clean install in CI environment. I'll raise a PR for `package-lock.json` on the repository.

**Comment by akumar074** [comment]
PR for package-lock.json: https://github.com/kiegroup/kogito-docs/pull/15

**Pull request merged**